x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -381,90 +380,43 @@ private static TestPhysicalOperationProviders testOperationProviders(CsvTestsDat
     }
 
     private ActualResults executePlan(BigArrays bigArrays) throws Exception {
-        var parsed = parser.createStatement(testCase.query);
+        LogicalPlan parsed = parser.createStatement(testCase.query);
         var testDataset = testsDataset(parsed);
+        LogicalPlan analyzed = analyzedPlan(parsed, testDataset);
 
-        String sessionId = "csv-test";
-        BlockFactory blockFactory = new BlockFactory(
-            bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST),
-            bigArrays,
-            ByteSizeValue.ofBytes(randomLongBetween(1, BlockFactory.DEFAULT_MAX_BLOCK_PRIMITIVE_ARRAY_SIZE.getBytes() * 2))
-        );
-        ExchangeSourceHandler exchangeSource = new ExchangeSourceHandler(between(1, 64), executor);
-        ExchangeSinkHandler exchangeSink = new ExchangeSinkHandler(blockFactory, between(1, 64), threadPool::relativeTimeInMillis);
-        LocalExecutionPlanner executionPlanner = new LocalExecutionPlanner(
-            sessionId,
-            "",
-            new CancellableTask(1, "transport", "esql", null, TaskId.EMPTY_TASK_ID, Map.of()),
-            bigArrays,
-            blockFactory,
-            randomNodeSettings(),
+        EsqlSession session = new EsqlSession(
+            getTestName(),
             configuration,
-            exchangeSource,
-            exchangeSink,
-            Mockito.mock(EnrichLookupService.class),
-            testOperationProviders(testDataset)
+            null,
+            null,
+            null,
+            functionRegistry,
+            new LogicalPlanOptimizer(new LogicalOptimizerContext(configuration)),
+            mapper,
+            TEST_VERIFIER
         );
-        //
-        // Keep in sync with ComputeService#execute
```

Comment:
The stuff in EsqlSession is now just called directly.

Reply:
This is nice, but I'd still preserve this comment and place it on top of `runPhase` in line 465; the stuff in there still needs to be close to how the real compute engine runs.